### PR TITLE
added changes

### PR DESCRIPTION
**Pull request by Rokdai**

### CODE REVIEW

File: src/components/app/app.js [hunk 1]
```diff
@@ -6,28 +6,35 @@ import ItemList from "../itemList";
...
   };
 
-  onHide() {
-    const { name, hide } = this.state;
+  onHide = () => {
+    const {
+      btn: { name, hide },
+    } = this.state;
     const newName = name === "hide" ? "show" : "hide";
     this.setState({
```
Comment (lesix96) [inline src/components/app/app.js:19]: this.setState((prevState, props) => ({
      btn: {
        name: prevState.name === "hide" ? "show" : "hide",
        hide: !prevState.hide,
      },
    }));

File: src/components/app/app.js [hunk 2]
```diff
@@ -6,28 +6,35 @@ import ItemList from "../itemList";
...
+    btn: { name: "hide", hide: true },
+    itemId: null,
   };
 
-  onHide() {
-    const { name, hide } = this.state;
+  onHide = () => {
+    const {
+      btn: { name, hide },
+    } = this.state;
```
Comment (lesix96) [inline src/components/app/app.js:17]: useless destructure

File: src/components/app/app.js [hunk 3]
```diff
@@ -6,28 +6,35 @@ import ItemList from "../itemList";
...
+  setItemId = (id) => {
+    this.setState({
+      itemId: id,
+    });
+  };
 
   render() {
-    const { name, hide } = this.state;
+    const {
+      btn: { name, hide },
```
Comment (lesix96) [inline src/components/app/app.js:35]: why did you create variables, that you have never used?

File: src/components/charDetails/charDetails.js [hunk 4]
```diff
@@ -1,30 +1,102 @@
...
-                        <span>1820</span>
-                    </li>
-                    <li className="list-group-item d-flex justify-content-between">
-                        <span className="term">Culture</span>
-                        <span>First</span>
-                    </li>
-                </ul>
-            </div>
-        );
+  setChar = () => {
```
Comment (lesix96) [inline src/components/charDetails/charDetails.js:24]: what is the purpose of this method?

File: src/components/charDetails/charDetails.js [hunk 5]
```diff
@@ -1,30 +1,102 @@
...
+            );
+          })}
+        </ul>
+      </>
+    );
+  };
+
+  checkContent = () => {
+    const { id } = this.props;
+    const { error } = this.state;
```
Comment (lesix96) [inline src/components/charDetails/charDetails.js:78]: if you use destructure, please do this at the beginning of the file and then use all of the variables, that you want to

File: src/components/charDetails/charDetails.js [hunk 6]
```diff
@@ -1,30 +1,102 @@
...
+    if (!id) {
+      return <span>Select character</span>;
+    } else if (error) {
+      return getComponent(<ErrorMessage name={"Not found"} />);
+    } else {
+      return this.setContent();
+    }
+  };
+
+  componentDidMount() {
```
Comment (lesix96) [inline src/components/charDetails/charDetails.js:89]: you should write lifecycle methods at the beginning of the file

File: src/components/itemList/itemList.js [hunk 7]
```diff
@@ -1,20 +1,87 @@
...
+    error: false,
+  };
+
+  setData = (page) => {
+    return gotServicePage({
+      url: characters,
+      page: 4,
+    })
+      .then((items) => {
+        let nameList = [];
```
Comment (lesix96) [inline src/components/itemList/itemList.js:25]: why let instead of const?

File: src/components/itemList/itemList.js [hunk 8]
```diff
@@ -1,20 +1,87 @@
...
+    } else if (error) {
+      return getComponent(<ErrorMessage name={"Characters not found"} />);
+    } else {
+      return this.setContent();
     }
-}
\ No newline at end of file
+  };
+
+  componentDidMount() {
```
Comment (lesix96) [inline src/components/itemList/itemList.js:80]: the same

File: src/components/itemList/itemList.js [hunk 9]
```diff
@@ -1,20 +1,87 @@
...
+              {name}
+            </li>
+          );
+        })}
+      </ul>
+    );
+  };
+
+  checkContent = () => {
+    const { loading, error } = this.state;
```
Comment (lesix96) [inline src/components/itemList/itemList.js:69]: the same

File: src/components/itemList/itemList.js [hunk 10]
```diff
@@ -1,20 +1,87 @@
...
-        );
+  setContent = () => {
+    const { itemList } = this.state;
+
+    return (
+      <ul className="item-list list-group">
+        {itemList.map(({ name, id }) => {
+          return (
+            <li
+              onClick={this.pushId}
```
Comment (lesix96) [inline src/components/itemList/itemList.js:55]: why didn't you pass id direkt as an argument into this function? why did you extract this id inside if pushId function? it is useless

File: src/components/randomChar/randomChar.js [hunk 11]
```diff
@@ -48,54 +44,56 @@ export default class RandomChar extends Component {
...
+  };
 
-    const charData = [
-      { name: "Gender", value: gender },
-      { name: "Born", value: born },
-      { name: "Died", value: died },
-      { name: "Culture", value: culture },
-    ];
+  showContent = () => {
+    const { char, charName } = this.state;
```
Comment (lesix96) [inline src/components/randomChar/randomChar.js:50]: the same

File: src/components/randomChar/randomChar.js [hunk 12]
```diff
@@ -48,54 +44,56 @@ export default class RandomChar extends Component {
...
+      return getComponent(<Spiner />);
+    } else if (error) {
+      return getComponent(<ErrorMessage name={"Fetch"} />);
+    } else {
+      return this.showContent();
+    }
+  };
 
-    return <div className="random-block rounded">{content}</div>;
+  componentDidMount() {
```
Comment (lesix96) [inline src/components/randomChar/randomChar.js:92]: the same

File: src/utils/getComponent.js [hunk 13]
```diff
@@ -0,0 +1,5 @@
+const getComponent = (component) => {
```
Comment (lesix96) [inline src/utils/getComponent.js:1]: useless method

File: src/components/app/app.js (hunk 1 above)
Comment (Rokdai) [reply src/components/app/app.js:19]: changed

File: src/components/app/app.js (hunk 2 above)
Comment (Rokdai) [reply src/components/app/app.js:17]: changed

File: src/components/app/app.js (hunk 3 above)
Comment (Rokdai) [reply src/components/app/app.js:35]: I used it bottom in render method

**Pull request merged**